Improve support for RTL languages

andreslucena opened the pull request:
#### :tophat: What? Why?

In #12131 we have a bug related to RTL languages. For making easy these kind of fixes, I improved the support for RTL languages, by detecting when there's language that needs it, as [described in this comment](https://github.com/decidim/decidim/issues/12131#issuecomment-1907759347).   

#### :pushpin: Related Issues

- Fixes #9870

#### Testing

1. Add a RTL locale in `development_app/config/initializers/decidim.rb`, for instance `ar` (Arabic) 
```ruby
  config.available_locales = %w(en ca es ar)
```
2. Add it to the organization with 
```console
bin/rails runner 'org = Decidim::Organization.first ; org.available_locales << "ar" ; org.save '
```
3. Restart the webserver
4. See that the direction of the whole page changes accordingly 

### :camera: Screenshots

![Screenshot of Decidim in Arabic with RTL support](https://github.com/decidim/decidim/assets/717367/60d034f8-b92c-4375-938c-986079888120)

:hearts: Thank you!

alecslupu reviewed (commented):
This PR works as advertised, but i have noticed several caveats: 

- Admin area does not support RTL. 
- Admin area does not work when there are missing translations in editor i got ` i18n['textStyle.heading'] is undefined`
- The Homepage: the See Proceses button remains in the right, near the heading. 
- Floating help button remains in the left, but when clicked pop-up is going to the right. 

![image](https://github.com/decidim/decidim/assets/105683/7090bad1-ddbf-4040-a41c-1dfd42e0f9d7)

Since is an improvement compared with what we had in the past, we could merge this as long as we create an issue to track the RTL problems. 

@andreslucena I am ok with the codebase, but i would like to know your input on the other issues discovered.

andreslucena commented:
I'll try to do some of those in #12390, as I want to separate the feature (making it easy to installations that have a RTL language to have support  to the direction without making overrides/hacks) vs the actual fixes at the CSS level, so we can backport the fixes to v0.28.

I'll mark these two PRs as draft and left you know when they're ready for another round

andreslucena commented:
>     Admin area does not support RTL.

Done in 3400d322af264d724674898fe2ba8f4d2a5b0439

>     Admin area does not work when there are missing translations in editor i got  i18n['textStyle.heading'] is undefined

Pending. I'll do another PR as the problem is that we don't fallback well here

>     The Homepage: the See Proceses button remains in the right, near the heading.

Done in #12390 in 3adb4eb4aa9b42cec914736150a991fc3ab2078a 

>     Floating help button remains in the left, but when clicked pop-up is going to the right.

Done in #12390 in 76e1454683d6ee69401dcb4426c6c88172a164f7

I've left the help button in the left but fixed the modal. Couldn't find an easy fix for the help text in the left, as I'd need to change the background too and it was not that easy to be honest.

alecslupu reviewed (changes requested):
@andreslucena , 2 things are bugging me. I guess we need to fix the menu. 

The process menu is not displayed properly 
Home ( OK ) 
![image](https://github.com/decidim/decidim/assets/105683/2faa1b82-39dc-4f05-a3d9-10de3fff50ab)

Process ( NOK ) 
![image](https://github.com/decidim/decidim/assets/105683/579aaef6-9613-4d18-9807-3cff2af33599)

The "See all processes" button is in right (Even though i could live with this)
![image](https://github.com/decidim/decidim/assets/105683/38f53808-82c2-4c30-9273-a363bc5bde52)

andreslucena commented:
@alecslupu mind that the fixes like `The "See all processes" button is in right (Even though i could live with this` are actually being handled in #12390, so we can backport those fixes and separate them from the current feature

alecslupu commented:
> @alecslupu mind that the fixes like `The "See all processes" button is in right (Even though i could live with this` are actually being handled in #12390, so we can backport those fixes and separate them from the current feature

I have not yet visited https://github.com/decidim/decidim/pull/12390 since it's draft state.

andreslucena commented:
> The process menu is not displayed properly

Fixed that in 12ec106f889993008640852e1fe1cfe7702b207d  in https://github.com/decidim/decidim/pull/12390

> The "See all processes" button is in right (Even though i could live with this) 

Fixed that in 3adb4eb4aa9b42cec914736150a991fc3ab2078a in https://github.com/decidim/decidim/pull/12390

> I have not yet visited https://github.com/decidim/decidim/pull/12390 since it's draft state.

I have some failing specs that I need to check. This PR (#12389) should work well and ready to be merged IMO.

alecslupu reviewed (approved):
since there are no new commits since my last review, and since the reported bugs are fixed in the scope of a second PR, LGTM.
